sklearn/feature_selection/_from_model.py

```diff
@@ -320,7 +320,7 @@ def _check_max_features(self, X):
                 "max_features",
                 Integral,
                 min_val=0,
-                max_val=n_features,
+                max_val=None,
```

Comment:
there's also no need to calculate `_num_features` here then, and therefore the method gets simplified quite a bit